integritybackend/numbers.py

```diff
@@ -35,65 +35,82 @@ def register(
+        else:
+            # nft_metadata = {
+            #     "name": asset_name,
+            #     "description": asset_description,
+            #     "external_url": f"ipfs://{asset_cid}",
+            #     "custom": custom,
+            # }
+            #
+            # registration_data = [
+            #     ("assetCid", asset_cid),
+            #     ("assetSha256", asset_sha256),
+            #     ("assetMimetype", asset_mime_type),
+            #     ("assetTimestampCreated", asset_timestamp_created),
+            #     ("custom", json.dumps(custom)),
+            #     ("nftContractAddress", nft_contract_address),
+            #     ("nftMetadata", json.dumps(nft_metadata)),
+            # ]
 
-            registration_data = [
-                ("assetCid", asset_cid),
-                ("assetSha256", asset_sha256),
-                ("assetMimetype", asset_mime_type),
-                ("assetTimestampCreated", asset_timestamp_created),
-                ("custom", json.dumps(custom)),
-                ("nftContractAddress", nft_contract_address),
-                ("nftMetadata", json.dumps(nft_metadata)),
-            ]
-
-        resp = requests.post(
-            _REGISTER,
-            headers={"Authorization": f"Bearer {config.NUMBERS_API_KEY}"},
-            data=registration_data,
-        )
-
-        if not resp.ok:
-            _logger.error(
-                f"Numbers registration failed: {resp.status_code} {resp.text}"
-            )
-            return None
+            raise NotImplementedError("Don't know how to handle nft contract address")
 
-        data = resp.json()
-        if data.get("response") is None:
-            _logger.warning(
-                "Numbers registration response did not have the 'response' field: %s",
-                resp.text,
-            )
-            return None
-        if data.get("status") != "success":
-            _logger.warning(
-                "Numbers registration didn't have success status: %s", resp.text
+        result = {}
+
+        for chain in chains:
+            if chain == "numbers":
+                server = "https://eo883tj75azolos.m.pipedream.net"
+            elif chain == "avalanche":
+                server = "https://eox7ryteolf6eh2.m.pipedream.net"
+            elif chain == "near":
+                server = "https://eof6acukpt2bka5.m.pipedream.net"
+            else:
+                raise NotImplementedError(f"Unknown chain {chain}")
+
+            resp = requests.post(
+                server,
+                headers={"Authorization": f"token {config.NUMBERS_API_KEY}"},
+                json=registration_data,
             )
-            return None
 
-        _logger.info(f"Numbers registration succeeded: {resp.text}")
-        return data["response"]
+            if not resp.ok:
+                _logger.error(
+                    f"Numbers registration failed: {resp.status_code} {resp.text}"
+                )
+                return None
```

Comment:
We can't just return here because a previous call may have succeeded. This function should always return a map with `chain` as index, and transmit what operations succeeded and failed.

Reply:
And then in the archive action, do we accept partial failures? I assumed any failure was unacceptable so that's why I had it fail fast like this. Will update.